src/App.tsx

```diff
@@ -1,24 +1,201 @@
+import { Footer } from './component/Footer/Footer';
+import { LoadType } from './types/LoadType';
+import { Header } from './component/Header/Header';
 
-const USER_ID = 0;
+const USER_ID = 6908;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [typeOfLoad, setTypeOfLoad] = useState<LoadType>(LoadType.All);
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [disableInput, setDisableInput] = useState(false);
+  const [loadId, setLoadId] = useState<number[]>([]);
+
+  const getTodosFromServer = async (userId: number) => {
+    try {
+      setIsLoading(true);
+      const getTodo = await getTodos(userId);
+
+      setTodos(getTodo);
+    } catch {
+      setErrorMessage('Unable to load todos');
+      setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const addNewTodo = useCallback(async (title: string) => {
+    if (!title.trim()) {
+      return 'Unable to add a todo';
+    }
+
+    setDisableInput(true);
+
+    const newTodo = {
+      id: 0,
+      title,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    try {
+      setErrorMessage('');
+      setTempTodo(newTodo);
+      const newTempTodo = await addTodo(newTodo);
+
+      setTodos(prev => ([
+        ...prev,
+        newTempTodo,
+      ]));
+    } catch {
+      setErrorMessage('Unable to add a todo');
+    } finally {
+      setDisableInput(false);
+      setTempTodo(null);
+    }
+
+    return setDisableInput(false);
+  }, []);
+
+  const deleteTodoById = useCallback(async (id: number) => {
+    try {
+      await deleteTodo(id);
+    } catch {
+      setErrorMessage('Unable to delete a todo');
+    }
+
+    setTodos(prev => prev.filter(todo => todo.id !== id));
+  }, []);
+
+  useEffect(() => {
+    getTodosFromServer(USER_ID);
+  }, []);
+
+  const activeTodos = useMemo(() => {
+    return todos.filter(({ completed }) => !completed).length;
+  }, [todos]);
+
+  const close = () => {
+    setErrorMessage('');
+  };
+
+  const onlyCompleted = todos.filter(todo => todo.completed);
+
+  const deleteCompleted = useCallback(async () => {
+    onlyCompleted.forEach(todo => {
+      deleteTodo(todo.id)
+        .then(() => {
+          setTodos(todos.filter(task => !task.completed));
+        });
+    });
+  }, [todos]);
+
+  const handleEdit = useCallback(async (id: number, value: Partial<Todo>) => {
+    setLoadId(prev => [
+      ...prev,
+      id,
+    ]);
+
+    try {
+      setTodos(prev => prev.map(todo => {
+        if (id !== todo.id) {
+          return todo;
+        }
+
+        return { ...todo, ...value };
+      }));
+    } catch {
+      setErrorMessage('Unable to update a todo');
+    } finally {
+      setLoadId(prev => prev.filter(num => num !== id));
+    }
+  }, []);
```

Comment:
this blocks try/catch/finally and async keyword make no sense, because for them you need some function which will return promise, in this function you should update the todo on server, not just in the app (make updateTodo function in api/todos.ts file)